src/core/mod.rs

```diff
@@ -101,6 +106,15 @@ impl BPFBuilder {
         Ok(self)
     }
 
+    /// Set the device to load the BPF program on, if applicable.
+    /// For example a network device if running XDP in hardware mode.
+    pub fn device<T: AsRef<str>>(self, device: T) -> Result<Self, BccError> {
+        Ok(Self {
```

Comment:
This should really be taking `mut self` and setting `self.device` to `Some` variant. Why are you constructing an entirely new `Self ` to return here? This doesn't align with the builder pattern

Reply:
Fixed! I think this was mostly a preference that I found taking `self` simpler and the builder pattern doesn't exactly specify that detail? :open_mouth:  Looking at the rest of the codebase it makes a lot of sense to use `mut self` to be consistent in any case.

Reply:
https://github.com/rust-unofficial/patterns/blob/master/patterns/builder.md

There's arguments for taking `&mut self` instead too as in https://doc.rust-lang.org/std/process/struct.Command.html

We're taking ownership of it anyway, and the types involved aren't `Copy` - so I just don't see a benefit to not taking `mut self`. Would have to check if there were any additional allocations with how you initially did this - but better to have the codebase consistent regardless.

Reply:
I'm not seeing this addressed in the current PR yet, maybe there's a commit you haven't pushed yet?

Reply:
Yup this is my bad, I missed updating that file. Will follow up on it!

Reply:
This is fixed now :)